pkg/connector/service.go

```diff
@@ -217,42 +220,33 @@ func (s *Service) RemoveProcessor(ctx context.Context, connectorID string, proce
 	return conn, err
 }
 
-func (s *Service) SetDestinationState(ctx context.Context, id string, state DestinationState) (Destination, error) {
+func (s *Service) SetState(ctx context.Context, id string, state any) (*Instance, error) {
 	conn, err := s.Get(ctx, id)
 	if err != nil {
 		return nil, err
 	}
 
-	dest, ok := conn.(Destination)
-	if !ok {
-		return nil, cerrors.Errorf("expected connector to be a Destination (ID: %s): %w", id, ErrInvalidConnectorType)
-	}
-
-	dest.SetState(state)
-	err = s.store.Set(ctx, id, dest)
-	if err != nil {
-		return nil, err
-	}
-
-	return dest, err
-}
-
-func (s *Service) SetSourceState(ctx context.Context, id string, state SourceState) (Source, error) {
-	conn, err := s.Get(ctx, id)
-	if err != nil {
-		return nil, err
+	switch state.(type) {
+	case SourceState:
+		if conn.Type != TypeSource {
+			return nil, cerrors.Errorf("expected connector to be a source (ID: %s): %w", id, ErrInvalidConnectorType)
```

Comment:
Makes sense! I think this happened because we were reusing an existing error `ErrInvalidConnectorType`, we need to rather create a separate error for an invalid connector state type 👍